src/book/mod.rs

```diff
@@ -254,25 +303,28 @@ impl MDBook {
 
                 let path = self.source_dir().join(&chapter_path);
                 info!("Testing file: {:?}", path);
-
-                // write preprocessed file to tempdir
-                let path = temp_dir.path().join(&chapter_path);
-                let mut tmpf = utils::fs::create_file(&path)?;
-                tmpf.write_all(ch.content.as_bytes())?;
-
-                let mut cmd = Command::new("./test");
-                cmd.arg(&path);
-
-                let output = cmd.output()?;
-
-                if !output.status.success() {
-                    failed = true;
-                    error!(
-                        "./test returned an error:\n\
-                        \n--- stdout\n{}\n--- stderr\n{}",
-                        String::from_utf8_lossy(&output.stdout),
-                        String::from_utf8_lossy(&output.stderr)
-                    );
+                let chapter_name = chapter_path.file_name().unwrap().to_str().unwrap();
+
+                for snippet in MDBook::enumerate_blocks(&ch.content, target_language) {
+                    info!("  testing snippet on line: {:?}", snippet.line);
+                    // write block to temp file
+                    let name = format!("{}_{}.lean", chapter_name, snippet.line);
+                    let path = temp_dir.path().join(&name);
+                    fs::write(&path, snippet.block);
+
+                    let mut cmd = Command::new(test_binary);
+                    cmd.arg(&path);
+                    let output = cmd.output()?;
+
+                    if !output.status.success() {
+                        failed = true;
+                        error!(
+                            "./test returned an error:\n\
```

Comment:
still mentions `./test`

Reply:
ah, thanks, fixed, it now uses test_binary, so the output is:

```
2021-09-29 14:12:07 [ERROR] (mdbook::book): lean returned an error:

--- stdout
C:\Users\clovett\AppData\Local\Temp\mdbook-gDsUzI\whatIsLean.md_11.lean:16:4: error: expected command

--- stderr
```